packages/components/src/card/README.md

```diff
@@ -0,0 +1,97 @@
+# Card
+
+Card provides a flexible and extensible content container.
+
+## Usage
+
+```jsx
+import { Card, CardBody } from '@wordpress/components';
+
+const Example = () => (
+	<Card>
+		<CardBody>...</CardBody>
+	</Card>
+);
+```
+
+## Props
+
+<!-- prettier-ignore -->
+Name | Type | Default | Description
+--- | --- | --- | ---
+`size` | `string` | `medium` | Determines the amount of padding within the card.
```

Comment:
@gziolo + @sirreal `garbage` and `nonsense` 😂 . There isn't a whitelisting or enum setup at this moment. Do we have a preferred way of doing this? Other systems use the `PropTypes` library, and TypeScript/Flow have type checking things.

Reply:
I'm not especially concerned that you can get surprising results when using strange properties. We might provide warnings for better developer experience with something like this:

```js
if ( process.env.NODE_ENV !== 'production' ) {
  if ( ! knownSizes.includes( size ) ) {
    console.warn(
      'An unrecognized `size` (%o) was provided to a `Card` component.\n' +
      'Expected one of: %o',
      size,
      knownSizes
    );
  }
}
```

I'm not sure whether this type of pattern is present around Gutenberg and it's not a big concern. We can come back after this PR is merged.

Reply:
See my related comment under a completely different issue https://github.com/WordPress/gutenberg/issues/11631#issuecomment-544858908 which I can nearly copy and paste here:

> I don't think we can enforce all the things we would love but we need to find a way to make it clear for folks using WP components that they aren't following the best practices. I'm inclined to adopt `warning` utility method which @diegohaz included in Reakit:
> https://github.com/reakit/reakit/blob/6efdb1e5c9e26724e47850a4e9e83a698311ebca/packages/reakit-utils/src/warning.ts#L3-L26
> based on FB version:
> https://github.com/facebook/react/blob/b87aabdfe1b7461e7331abb3601d9e6bb27544bc/packages/shared/warningWithoutStack.js#L17-L42
> This would make it nearly impossible to miss those warnings when using the development mode.

In this case, I think it might be too noisy to show warnings on JS console. Instead, we could
use `@enum` tag from JSDoc to list all possible values as explained in their docs: https://jsdoc.app/tags-enum.html – this will ensure that folks using `wp-scripts lint-js` will see warnings errors when writing code comments (this includes Gutenberg). In the long run, we might switch to TypeScript which would be still opt-in so it wouldn't solve the issue for all developers.

Again, in terms of warnings, we should decide about the strategy where to use them so they would be helpful for developers rather than too noisy and thus ignored. My initial idea was to use warnings as accessibility guidelines which is something you can't provide for every individual component.

@youknowriad, you might have some good thoughts on it.